mungegithub/mungers/release-note-label.go

```diff
@@ -144,20 +144,15 @@ func (r *ReleaseNoteLabel) Munge(obj *github.MungeObject) {
 		return
 	}
 
+	releaseNoteLabelNeededAdded := false
 	if !obj.HasLabel(releaseNoteLabelNeeded) {
 		obj.AddLabel(releaseNoteLabelNeeded)
+		releaseNoteLabelNeededAdded = true
 	}
 
-	if !obj.HasLabel(lgtmLabel) {
-		return
+	if releaseNoteLabelNeededAdded && obj.HasLabel(lgtmLabel) {
+		obj.WriteComment(releaseNoteBody)
 	}
-
-	if obj.HasLabel(doNotMergeLabel) {
-		return
-	}
-
-	obj.WriteComment(releaseNoteBody)
-	obj.AddLabel(doNotMergeLabel)
```

Comment:
Two possible solutions:
- Use `!obj.HasLabel(releaseNoteLabelNeeded)` as the tracking mechanism and as soon as we poll for release note, if the PR doesn't have the right label, add the comment in the PR right away.
- Only add the note at LGTM time, but remove the LGTM if we don't have the right releasenote process followed. If we did this, we could simplify things further by not needing to make release-note-label-needed a blocking label, because removing the LGTM would have the same effect.